lib/pbench/server/api/resources/upload_api.py

```diff
@@ -237,39 +262,44 @@ def put(self, filename: str):
             # First write the .md5
             try:
                 md5_full_path.write_text(f"{md5sum} {filename}\n")
-            except Exception:
+            except Exception as exc:
                 try:
                     md5_full_path.unlink(missing_ok=True)
-                except Exception as exc:
+                except Exception as md5_exc:
                     self.logger.error(
-                        "Failed to remove .md5 {} when trying to clean up: {}",
+                        "Failed to remove .md5 {} when trying to clean up: '{}'",
                         md5_full_path,
-                        exc,
+                        md5_exc,
                     )
-                self.logger.exception("Failed to write .md5 file, '{}'", md5_full_path)
-                raise
+                self.logger.error(
+                    "Failed to write .md5 file, '{}': '{}'", md5_full_path, exc
+                )
+                abort(HTTPStatus.INTERNAL_SERVER_ERROR, message="INTERNAL ERROR")
 
             # Then create the final filename link to the temporary file.
             try:
                 os.link(ofp.name, tar_full_path)
-            except Exception:
+            except Exception as exc:
                 try:
                     md5_full_path.unlink()
-                except Exception as exc:
+                except Exception as md5_exc:
                     self.logger.error(
                         "Failed to remove .md5 {} when trying to clean up: {}",
                         md5_full_path,
-                        exc,
+                        md5_exc,
                     )
-                self.logger.exception(
-                    "Failed to rename tar ball '{}' to '{}'", ofp.name, md5_full_path,
+                self.logger.error(
+                    "Failed to rename tar ball '{}' to '{}': '{}'",
+                    ofp.name,
+                    md5_full_path,
+                    exc,
                 )
-                raise
+                abort(HTTPStatus.INTERNAL_SERVER_ERROR, message="INTERNAL ERROR")
```

Comment:
What is the default behavior for an unhandled exception?  Is it `500/INTERNAL ERROR`?  If so, maybe the re-raise wasn't such a bad idea....  That is, you indicated that we already get a log message from the `abort()`...so, perhaps there is no need for the inner `try` block at _all_ and the outer `except` block doesn't need to do anything other than invoke `md5_full_path.unlink()` and `raise` (or call `abort()`)?

Reply:
We need the inner `try` block to prevent any exception from the `unlink()` from being raised at all.  We just want to "note" that something happened while handling the first exception, and then let the first exception propagate.

Reply:
I was thinking more generally:  it seems like we have a lot of places where we catch exceptions just so that we can log them and then call `abort()` with a `500` error...if it turned out that Flask's default reaction to an unhandled exception was to log it and abort with a `500` error, then we could get rid of a lot of code....

Reply:
Potentially.  There are ways to add generic error handling (see https://python.plainenglish.io/python-web-development-with-flask-error-handling-36349a7f82c).

This specific problem is that `abort()` does not give us a way to add color to what is logged.

But in another commit, we could re-work this code that it only raises an exception class we know about where, we provide a response code, message (optional), and logging context.

Reply:
Tracked in #2324.